lib/tracing-limit/src/lib.rs

```diff
@@ -15,13 +14,16 @@ use tracing_subscriber::layer::{Context, Layer};
 extern crate tracing;
 
 const RATE_LIMIT_SECS_FIELD: &str = "internal_log_rate_secs";
-const COMPONENT_NAME_FIELD: &str = "component_name";
 const MESSAGE_FIELD: &str = "message";
 
+/// Keys in RATE_LIMIT_SPAN_GROUPS will cause events to be independently rate limited by the values
+/// for these keys
+const RATE_LIMIT_SPAN_GROUPS: &[&str] = &["component_name", "vrl_line_number", "vrl_position"];
+
 #[derive(Eq, PartialEq, Hash, Clone)]
 struct RateKeyIdentifier {
     callsite: Identifier,
-    component_name: Option<String>,
+    rate_limit_key_values: Vec<(String, TraceValue)>,
```

Comment:
I actually realized there is an issue here that if the span keys are associated in a differing order, then the identifier won't match. I don't think this would actually happen in practice, but I'll try to swap this with a data structure that doesn't care about ordering (maybe a`HashMap` or `HashSet`).

Reply:
I think it's safe to assume that the order won't change, could go further to make this an `[Option<TraceValue>; 3]` too, because the keys are static at the moment.

I also think it is possible to use `&'static str` here.

Reply:
I ended up swapping this back to a concrete struct in https://github.com/timberio/vector/pull/6615/commits/acb520b234b7b3471a5f6cb123a5fce0fd7bdb15

I think this has the following advantages:

* Doesn't rely on span ordering (e.g. if component_name appears before
  vrl_line_number sometimes and after others)
* Allows overriding by inner spans of other span values for these keys